sdk/ai/azure-ai-agents/samples/agents_async/sample_agents_deep_research_async.py

```diff
@@ -42,51 +43,171 @@
 from azure.identity.aio import DefaultAzureCredential
 
 
+def convert_citations_to_superscript(markdown_content):
+    """
+    Convert citation markers in markdown content to HTML superscript format.
+
+    This function finds citation patterns like 【78:12†source】 and converts them to
+    HTML superscript tags <sup>12</sup> for better formatting in markdown documents.
+    It also consolidates consecutive citations by sorting and deduplicating them.
+
+    Args:
+        markdown_content (str): The markdown content containing citation markers
+
+    Returns:
+        str: The markdown content with citations converted to HTML superscript format
+    """
+    # Pattern to match 【number:number†source】
+    pattern = r"【\d+:(\d+)†source】"
```

Comment:
Let us avoid Unicode characters literal and compile a regex:
```python
pattern = re.compild("\u3010\d+:(\d+)†source\u3011")
```
then we can use shorter substitution:
```python
pattern.sub(replacement, markdown_content)
```